pkg/azure/blob/credentials_test.go

```diff
@@ -0,0 +1,132 @@
+package blob
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"get.porter.sh/plugin/azure/pkg/azure/azureconfig"
+	"github.com/hashicorp/go-hclog"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGet_GetCredentials(t *testing.T) {
+	testcases := []struct {
+		name         string
+		envVarsToSet map[string]string
+		config       *azureconfig.Config
+		testfunc     func(t *testing.T, envVarsToSet map[string]string, config *azureconfig.Config, logger hclog.Logger)
```

Comment:
A test pattern that we use often in Porter would help simplify this a lot:

```go 
testcases := []struct {
    name         string
    envVarsToSet map[string]string
    config       *azureconfig.Config
    wantError   string
}

// ...

for k, v := range tc.envVarsToSet {
  os.Setenv(k, v)
}

_, err := GetCredentials(*config, logger)
if wantError == "" {
  require.NoError(t, err, "GetCredentials should have not returned an error")
} else {
  require.Error(t, err, "GetCredentials should have returned an error")
  assert.Contains(t, err.Error(), tc.wantError, "Unexpected error returned from GetCredentials")
}
```

This keeps all the logic _inside_ the test, ensures each test case is tested the same way, and removes all those test helper functions.

---

I also noticed that even in your test case for valid connection string that you don't check the values returned to ensure they are correct. What works well is to add a field to the testcase for `wantCred` and assert they are equal when `wantError` is empty.